pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -357,6 +358,76 @@ func TestOutboundListenerTCPWithVS(t *testing.T) {
 	}
 }
 
+func TestOutboundListenerForHeadlessServices(t *testing.T) {
+	_ = os.Setenv("PILOT_ENABLE_FALLTHROUGH_ROUTE", "false")
+
+	defer func() { _ = os.Unsetenv("PILOT_ENABLE_FALLTHROUGH_ROUTE") }()
+
+	svc := buildServiceWithPort("test.com", 9999, protocol.TCP, tnow)
+	svc.Attributes.ServiceRegistry = string(serviceregistry.KubernetesRegistry)
+	svc.Resolution = model.Passthrough
+	services := []*model.Service{svc}
+
+	p := &fakePlugin{}
+	tests := []struct {
+		name                      string
+		instances                 []*model.ServiceInstance
+		numListenersOnServicePort int
+	}{
+		{
+			name: "less than 5 instances",
+			instances: []*model.ServiceInstance{
+				buildServiceInstance(services[0], "10.10.10.10"),
+				buildServiceInstance(services[0], "11.11.11.11"),
+				buildServiceInstance(services[0], "12.11.11.11"),
+			},
+			numListenersOnServicePort: 3,
+		},
+		{
+			name: "more than 5 instances",
+			instances: []*model.ServiceInstance{
+				buildServiceInstance(services[0], "10.10.10.10"),
+				buildServiceInstance(services[0], "11.11.11.11"),
+				buildServiceInstance(services[0], "12.11.11.11"),
+				buildServiceInstance(services[0], "13.11.11.11"),
+				buildServiceInstance(services[0], "14.11.11.11"),
+				buildServiceInstance(services[0], "15.11.11.11"),
+			},
+			numListenersOnServicePort: 6,
```

Comment:
we dont need this kind of check .. <5 or >5. We generate these listeners regardless of the count of pods.

Reply:
fair point